src/configuration/mod.rs

```diff
@@ -1,56 +1,339 @@
+        self.socket_path = Some(socket_path);
+        self
+    }
+
+    pub fn with_batching_options(mut self, batching_options: dogstatsd::BatchingOptions) -> Self {
+        self.batching_options = Some(batching_options);
+        self
+    }
+
+    pub fn to_addr(&self) -> &str {
         self.to_addr.as_str()
     }
 
-    fn from_addr(&self) -> &str {
+    pub fn from_addr(&self) -> &str {
         self.from_addr.as_str()
     }
 
-    fn namespace(&self) -> &str {
+    pub fn namespace(&self) -> &str {
         self.namespace.as_str()
     }
 
-    fn default_tags(&self) -> Vec<String> {
-        vec![]
+    pub fn default_tags(&self) -> Vec<String> {
+        self.tags.clone()
     }
 
-    fn socket_path(&self) -> Option<String> {
+    pub fn socket_path(&self) -> Option<String> {
         self.socket_path.clone()
     }
 
-    fn batching_options(&self) -> Option<dogstatsd::BatchingOptions> {
+    pub fn batching_options(&self) -> Option<dogstatsd::BatchingOptions> {
         self.batching_options
     }
+
+    pub fn take_tracker_config(&mut self) -> TagTrackerConfiguration {
+        std::mem::replace(&mut self.tracker, TagTrackerConfiguration::new())
+    }
+}
+
+impl From<Configuration> for dogstatsd::Options {
+    fn from(value: Configuration) -> Self {
+        dogstatsd::Options::new(
+            value.from_addr(),
+            value.to_addr(),
+            value.namespace(),
+            value.default_tags(),
+            value.socket_path(),
+            value.batching_options(),
+        )
+    }
+}
+
+/// Represent an environment in which the datadog client runs.
+/// This is useful for enforcing rules based on environment for every application that uses the library.
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub enum Environment {
+    Dev,
+    Staging,
+    Production,
+}
+
+impl Environment {
+    /// Returns the string representation of the environment.
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            Environment::Dev => "dev",
+            Environment::Staging => "staging",
+            Environment::Production => "production",
+        }
+    }
+}
+
+impl FromStr for Environment {
+    type Err = PrimaDatadogError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "dev" => Ok(Self::Dev),
+            "staging" => Ok(Self::Staging),
+            "production" => Ok(Self::Production),
+            _ => Err(PrimaDatadogError::WrongEnvironmentDefinition),
+        }
+    }
+}
+
+impl Display for Environment {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
+
+#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
+#[cfg(feature = "serde")]
+impl serde::Serialize for Environment {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.as_str())
+    }
+}
```

Comment:
Is there any reason to have serialize impls here, or is it just a "why not" situation?